dbms/src/Storages/KVStore/KVStore.cpp

```diff
@@ -375,33 +374,25 @@ void KVStore::setRegionCompactLogConfig(UInt64 rows, UInt64 bytes, UInt64 gap, U
 
 void KVStore::persistRegion(
     const Region & region,
-    std::optional<const RegionTaskLock *> region_task_lock,
+    const RegionTaskLock & region_task_lock,
     PersistRegionReason reason,
     const char * extra_msg) const
 {
     RUNTIME_CHECK_MSG(
         region_persister,
         "try access to region_persister without initialization, stack={}",
         StackTrace().toString());
-    if (region_task_lock.has_value())
-    {
-        auto reason_id = magic_enum::enum_underlying(reason);
-        std::string caller = fmt::format("{} {}", PersistRegionReasonMap[reason_id], extra_msg);
-        LOG_INFO(
-            log,
-            "Start to persist {}, cache size: {} bytes for `{}`",
-            region.getDebugString(),
-            region.dataSize(),
-            caller);
-        region_persister->persist(region, *region_task_lock.value());
-        LOG_DEBUG(log, "Persist {} done", region.toString(false));
-    }
-    else
-    {
-        LOG_INFO(log, "Try to persist {}", region.toString(false));
-        region_persister->persist(region);
-        LOG_INFO(log, "After persisted {}, cache {} bytes", region.toString(false), region.dataSize());
-    }
+
+    auto reason_id = magic_enum::enum_underlying(reason);
+    std::string caller = fmt::format("{} {}", PersistRegionReasonMap[reason_id], extra_msg);
+    LOG_INFO(
+        log,
+        "Start to persist {}, cache size: {} bytes for `{}`",
+        region.getDebugString(),
+        region.dataSize(),
+        caller);
+    region_persister->persist(region, region_task_lock);
+    LOG_DEBUG(log, "Persist {} done, cache size: {} bytes", region.toString(false), region.dataSize());
```

Comment:
Use `region size` for `region.dataSize()`. Though it's quite misleading that we have both `approx_mem_cache_bytes` and `region.dataSize()`. I think maybe we optimize it later.